test/packager.spec.ts

```diff
@@ -0,0 +1,1509 @@
+        ...baseOpts,
+      };
+
+      const paths = await packager(opts);
+      expect(paths).toHaveLength(1);
+      expect(paths[0]).toEqual(
+        path.join(
+          opts.out!,
+          generateFinalBasename({
+            name: opts.name,
+            platform: process.platform,
+            arch: getHostArch(),
+          }),
+        ),
+      );
+      // Create a dummy file to detect whether the output directory is replaced in subsequent runs
+      const testPath = path.join(paths[0], 'test.txt');
+      await fs.writeFile(testPath, 'test');
+      // Second run without overwrite should be skipped
+      await packager(opts);
+      expect(fs.existsSync(testPath)).toBe(true);
+      // Third run with overwrite should replace the output directory
+      await packager({ ...opts, overwrite: true });
+      expect(fs.existsSync(testPath)).toBe(false);
+    },
+  );
+
+  it('defaults the out directory to the current working directory', async ({
+    baseOpts,
+  }) => {
+    const opts = {
+      ...baseOpts,
+      out: undefined,
+    };
+    const paths = await packager(opts);
+    expect(paths).toHaveLength(1);
+    expect(paths[0]).toEqual(
+      path.join(
+        process.cwd(),
+        generateFinalBasename({
+          name: opts.name,
+          platform: process.platform,
+          arch: getHostArch(),
+        }),
+      ),
+    );
+    expect(paths[0]).toBeDirectory();
+    await fs.rm(paths[0], { recursive: true, force: true });
+  });
+
+  it('can package with platform/arch set', async ({ baseOpts }) => {
+    const opts: Options = {
+      ...baseOpts,
+      platform: 'linux',
+      arch: 'x64',
+    };
+
+    const paths = await packager(opts);
+    expect(paths).toHaveLength(1);
+    expect(paths[0]).toEqual(
+      path.join(
+        opts.out!,
+        generateFinalBasename({
+          platform: opts.platform as string,
+          arch: opts.arch as string,
+          name: opts.name,
+        }),
+      ),
+    );
+  });
+
+  it('can package with tmpdir disabled', async ({ baseOpts }) => {
+    const opts: Options = {
+      ...baseOpts,
+      tmpdir: false,
+    };
+    const paths = await packager(opts);
+    expect(paths).toHaveLength(1);
+    expect(paths[0]).toBeDirectory();
+  });
+
+  it('preserves symlinks with derefSymlinks disabled', async ({ baseOpts }) => {
+    const opts: Options = {
+      ...baseOpts,
+      derefSymlinks: false,
+      platform: 'linux',
+      arch: 'x64',
+    };
+
+    const src = path.join(opts.dir, 'main.js');
+    const dest = path.join(opts.dir, 'main-link.js');
+    await fs.symlink(src, dest);
+
+    const paths = await packager(opts);
+    expect(paths).toHaveLength(1);
+
+    const destLink = path.join(paths[0], 'resources', 'app', 'main-link.js');
+    expect(destLink).toBeSymlink();
+
+    await fs.rm(dest, { force: true });
+  });
+
+  // FIXME: This flakes with ENOTEMPTY: directory not empty
+  it.runIf(process.platform === 'darwin').skip(
```

Comment:
I ended up skipping this test because I think this may be a valid bug with concurrency in universal packaging. Will need to investigate a bit more.